tests/cupyx_tests/scipy_tests/interpolate_tests/test_polyint.py

```diff
@@ -0,0 +1,256 @@
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs)
+        P.set_yi(ys)
+        return P(test_xs)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_append(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind == 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        test_xs = xp.linspace(-5, 5, 100, dtype=dtype)
+        xs = xp.linspace(-5, 5, 5, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs[:3], ys[:3])
+        P.add_xi(xs[3:], ys[3:])
+        return P(test_xs)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-5, rtol=1e-5)
+    def test_vector(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind == 'u':
+            pytest.skip()
+        xs = xp.array([0, 1, 2], dtype=dtype)
+        ys = xp.array([[0, 1], [1, 0], [2, 1]], dtype=dtype)
+        test_xs = xp.linspace(-1, 3, 100, dtype=dtype)
+        P = scp.interpolate.BarycentricInterpolator(xs, ys)
+        return P(test_xs)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_shapes_scalarvalue_1(self, xp, scp, dtype):
+        true_poly = xp.poly1d([-2, 3, 5, 1, -3])
+        xs = xp.linspace(-1, 10, 10, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs, ys)
+        return xp.shape(P(0))
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_shapes_scalarvalue_2(self, xp, scp, dtype):
+        true_poly = xp.poly1d([-2, 3, 5, 1, -3])
+        xs = xp.linspace(-1, 10, 10, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs, ys)
+        return xp.shape(P(xp.array(0, dtype=dtype)))
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_shapes_scalarvalue_3(self, xp, scp, dtype):
+        true_poly = xp.poly1d([-2, 3, 5, 1, -3])
+        xs = xp.linspace(-1, 10, 10, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs, ys)
+        return xp.shape(P([0]))
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_shapes_scalarvalue_4(self, xp, scp, dtype):
+        true_poly = xp.poly1d([-2, 3, 5, 1, -3])
+        xs = xp.linspace(-1, 10, 10, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.BarycentricInterpolator(xs, ys)
+        return xp.shape(P([0, 1]))
+
+    @testing.for_all_dtypes(no_bool=True, no_float16=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-6, rtol=1e-6)
+    def test_shapes_vectorvalue_1(self, xp, scp, dtype):
```

Comment:
`test_shapes_vectorvalue_*` do not test the shapes.

Reply:
changed the name, since we are already testing the shapes below.